clients/js/src/wrap.ts

```diff
@@ -47,22 +50,151 @@ const getOwnerFromAccount = async (
   return accountInfo.value.owner;
 };
 
-export interface ExecuteWrapArgs {
+interface TxBuilderArgs {
+  payer: KeyPairSigner | Address;
+  unwrappedTokenAccount: Address;
+  escrowAccount: Address;
+  wrappedTokenProgram: Address;
+  amount: bigint | number;
+  wrappedMint: Address;
+  wrappedMintAuthority: Address;
+  blockhash: {
+    blockhash: Blockhash;
+    lastValidBlockHeight: bigint;
+  };
+  transferAuthority: Address | TransactionSigner;
+  unwrappedMint: Address;
+  recipientWrappedTokenAccount: Address;
+  unwrappedTokenProgram: Address;
+  multiSigners?: TransactionSigner[];
+}
+
+// Used to collect signatures
+export const multisigOfflineSignWrap = async (args: Required<TxBuilderArgs>) => {
+  return buildWrapTransaction(args);
+};
+
+const messageBytesEqual = (
+  results: Awaited<ReturnType<typeof multisigOfflineSignWrap>>[],
+): boolean => {
+  // If array has only one element, return true
+  if (results.length === 1) {
+    return true;
+  }
+
+  // Use the first result as reference
+  const reference = results[0];
+  if (!reference) throw new Error('No transactions in input');
+
+  // Compare each result with the reference
+  for (let i = 1; i < results.length; i++) {
+    const current = results[i];
+    if (!current) throw new Error('Nullish entry in signature results array');
+
+    // Compare messageBytes
+    if (reference.messageBytes.length !== current.messageBytes.length) {
+      return false;
+    }
+
+    for (let j = 0; j < reference.messageBytes.length; j++) {
+      if (reference.messageBytes[j] !== current.messageBytes[j]) {
+        return false;
+      }
+    }
```

Comment:
We have a function `containsBytes` in Kit that is very similar to this and might be easier. You'll need to check the lengths are equal still, but it'd replace that second loop. I think you could replace this with:

```typescript
const matches = reference.messageBytes.length === current.messageBytes.length && containsBytes(reference.messageBytes, current.messageBytes, 0)
if(!matches) return false;
```

The 0 there is an `offset` param, you'd want to compare from the start of the byte array